cli/tests/rsa_signature.go

```
@@ -0,0 +1,380 @@
+		cfg.SSVOptions.ValidatorOptions.BeaconNetwork = networkConfig.Beacon.GetNetwork()
+		cfg.SSVOptions.ValidatorOptions.ShareEncryptionKeyProvider = nodeStorage.GetPrivateKey
+
+		validatorCtrl := validator.NewController(logger, cfg.SSVOptions.ValidatorOptions)
+		cfg.SSVOptions.ValidatorController = validatorCtrl
+
+		if err := p2pNetwork.Setup(logger); err != nil {
+			logger.Fatal("failed to setup network", zap.Error(err))
+		}
+		if err := p2pNetwork.Start(logger); err != nil {
+			logger.Fatal("failed to start network", zap.Error(err))
+		}
+		messageRoute := &messageRouter{
+			logger: logger,
+			ch:     make(chan *queue.DecodedSSVMessage, 1024),
+		}
+		p2pNetwork.UseMessageRouter(messageRoute)
+		go func() {
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+			ch := messageRoute.GetMessageChan()
+			sendMessageCount := 0
+			for {
+				select {
+				case <-ctx.Done():
+					logger.Debug("router message handler stopped")
+					return
+				case incomingMessage := <-ch:
+					ssvMessage := incomingMessage.SSVMessage
+					if mm, ok := incomingMessage.Body.(*spectypes.SignedPartialSignatureMessage); ok {
+						// Seed the random number generator
+						// #nosec G404
+						seed := time.Now().UnixNano() // Using current time as a seed for demonstration
+						// #nosec G404
+						localRand := rand.New(rand.NewSource(seed))
+
+						// Generate a random number, for example between 0 and 9
+						// #nosec G404
+						dynamicValue := localRand.Intn(10) // This will give you a random integer between 0 and 9
+
+						// Using fmt.Sprintf to insert the dynamic value into the string
+						signatureString := fmt.Sprintf("sVV0fsvqQlqliKv/ussGIatxpe8LDWhc8uoaM5WpjbiYvvxUr1eCpz0ja7UT%dPGNDdmoGi6xbMC1g/ozhAt4uCdpy0Xdfqbv", dynamicValue)
+
+						// Converting the formatted string to a byte slice
+						signature := []byte(signatureString)
+						mm.Signature = signature
+						incomingMessage.Data, err = mm.Encode()
```

Error is not checked